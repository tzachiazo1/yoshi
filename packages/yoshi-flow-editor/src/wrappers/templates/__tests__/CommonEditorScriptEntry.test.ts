import commonEditorScriptEntry from '../CommonEditorScriptEntry';

describe('CommonEditorScriptEntry template', () => {
  it('generates correct template with entry editorScript file for OOI components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      artifactId: 'some-app',
      experimentsConfig: {
        scope: 'test-scope',
      },
      editorScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/editorScript.js',
      sentry: null,
      controllersMeta: [],
      shouldUseAppBuilder: false,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template with entry editorScript file for OOI components using sentry', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      experimentsConfig: null,
      sentry: {
        DSN: 'kkk',
        id: '1',
        teamName: 'some-team',
        projectName: 'some-project',
      },
      artifactId: 'some-app',
      editorScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/editorScript.js',
      controllersMeta: [],
      shouldUseAppBuilder: false,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template without entry editorScript file for OOI components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: null,
      experimentsConfig: null,
      sentry: null,
      artifactId: 'some-app',
      editorScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/editorScript.js',
      controllersMeta: [],
      shouldUseAppBuilder: false,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template with entry editorScript file for app builder components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      artifactId: 'some-app',
      experimentsConfig: {
        scope: 'test-scope',
      },
      editorScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/editorScript.js',
      sentry: null,
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/a/editor.controller.ts',
          id: 'FIRST_ID',
          widgetType: 'STUDIO_WIDGET',
        },
      ],
      shouldUseAppBuilder: true,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template with entry editorScript file for multiple app builder components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      artifactId: 'some-app',
      experimentsConfig: {
        scope: 'test-scope',
      },
      editorScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/editorScript.js',
      sentry: null,
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/a/editor.controller.ts',
          id: 'FIRST_ID',
          widgetType: 'STUDIO_WIDGET',
        },
        {
          controllerFileName: 'project/src/components/c/editor.controller.ts',
          id: 'SECOND_ID',
          widgetType: 'STUDIO_WIDGET',
        },
        {
          controllerFileName: 'project/src/components/d/editor.controller.ts',
          id: 'THIRD_ID',
          widgetType: 'STUDIO_WIDGET',
        },
      ],
      shouldUseAppBuilder: true,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });
});
