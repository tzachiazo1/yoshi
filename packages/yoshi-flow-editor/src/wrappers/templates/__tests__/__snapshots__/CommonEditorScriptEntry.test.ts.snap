// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for OOI components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  

  
  var editorReadyWrapper = require('yoshi-flow-editor-runtime/build/editorScript.js').editorReadyWrapper;
  var editorReady = editorScriptEntry.editorReady;

  var sentry = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  if (editorReady) {
    editorReady = editorReadyWrapper(editorReady, sentry, experimentsConfig, 'some-app');
  }

  module.exports = editorScriptEntry.default || {
    ...editorScriptEntry,
    editorReady,
  };
  
"
`;

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for OOI components using sentry 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  

  
  var editorReadyWrapper = require('yoshi-flow-editor-runtime/build/editorScript.js').editorReadyWrapper;
  var editorReady = editorScriptEntry.editorReady;

  var sentry = {
    DSN: 'kkk',
    id: '1',
    projectName: 'some-project',
    teamName: 'some-team',
  };

  var experimentsConfig = null;

  if (editorReady) {
    editorReady = editorReadyWrapper(editorReady, sentry, experimentsConfig, 'some-app');
  }

  module.exports = editorScriptEntry.default || {
    ...editorScriptEntry,
    editorReady,
  };
  
"
`;

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for app builder components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  
    if (editorScriptEntry.default) {
      module.exports = editorScriptEntry.default;
    } else {
      var editorScriptBuilder = require('@wix/bob-widget-services').editorScriptBuilder;

      var builder = editorScriptBuilder();
      if (editorScriptEntry.editorReady) {
        builder = builder.withEditorReady(editorScriptEntry.editorReady);
      }
      if (editorScriptEntry.appManifest) {
        builder = builder.withAppManifest(editorScriptEntry.appManifest);
      }
      if (editorScriptEntry.eventHandler) {
        builder = builder.withEventHandler(editorScriptEntry.eventHandler);
      }
      
      var userController_0 = require('project/src/components/a/editor.controller.ts');
      builder = builder.withWidget(userController_0.default || userController_0);

      module.exports = builder.build();
  }
  

  
"
`;

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for multiple app builder components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  
    if (editorScriptEntry.default) {
      module.exports = editorScriptEntry.default;
    } else {
      var editorScriptBuilder = require('@wix/bob-widget-services').editorScriptBuilder;

      var builder = editorScriptBuilder();
      if (editorScriptEntry.editorReady) {
        builder = builder.withEditorReady(editorScriptEntry.editorReady);
      }
      if (editorScriptEntry.appManifest) {
        builder = builder.withAppManifest(editorScriptEntry.appManifest);
      }
      if (editorScriptEntry.eventHandler) {
        builder = builder.withEventHandler(editorScriptEntry.eventHandler);
      }
      
      var userController_0 = require('project/src/components/a/editor.controller.ts');
      builder = builder.withWidget(userController_0.default || userController_0);
  
      var userController_1 = require('project/src/components/c/editor.controller.ts');
      builder = builder.withWidget(userController_1.default || userController_1);
  
      var userController_2 = require('project/src/components/d/editor.controller.ts');
      builder = builder.withWidget(userController_2.default || userController_2);

      module.exports = builder.build();
  }
  

  
"
`;

exports[`CommonEditorScriptEntry template generates correct template without entry editorScript file for OOI components 1`] = `
"
  var editorScriptEntry = { editorReady: function {} };

  

  
  var editorReadyWrapper = require('yoshi-flow-editor-runtime/build/editorScript.js').editorReadyWrapper;
  var editorReady = editorScriptEntry.editorReady;

  var sentry = null;

  var experimentsConfig = null;

  if (editorReady) {
    editorReady = editorReadyWrapper(editorReady, sentry, experimentsConfig, 'some-app');
  }

  module.exports = editorScriptEntry.default || {
    ...editorScriptEntry,
    editorReady,
  };
  
"
`;
